src/main/scala/com/microsoft/hyperspace/index/IndexSignatureProvider.scala

```diff
@@ -37,15 +38,19 @@ class IndexSignatureProvider extends LogicalPlanSignatureProvider {
   /**
    * Generate the signature of logical plan.
    *
-   * @param logicalPlan logical plan.
+   * @param logicalPlan logical plan of data frame.
+   * @param targetFiles list of file paths used to calculate signature
    * @return signature, if both [[FileBasedSignatureProvider]] and [[PlanSignatureProvider]]
    *         can generate signature for the logical plan; Otherwise None.
    */
-  def signature(logicalPlan: LogicalPlan): Option[String] = {
-    fileBasedSignatureProvider.signature(logicalPlan).flatMap { f =>
+  override def signature(
+      logicalPlan: LogicalPlan,
+      targetFiles: Option[Set[Path]] = None): Option[String] = {
+    fileBasedSignatureProvider.signature(logicalPlan, targetFiles).flatMap { f =>
       planSignatureProvider.signature(logicalPlan).map { p =>
```

Comment:
```suggestion
      planSignatureProvider.signature(logicalPlan, targetFiles).map { p =>
```

We can let the planSignatureProvider decide whether to use or ignore the targetFiles (which it ignores currently)